From 654c744b403dc939569eb1687da06d9be6626c98 Mon Sep 17 00:00:00 2001
From: Dave Daynard <nardholio@gmail.com>
Date: Tue, 2 Sep 2014 19:14:08 -0400
Subject: [PATCH 041/206] Camera2: Headset shutter mode

Use play button on headset to take a picture/video/panorama
Use media next/previous for zoom

Change-Id: I25c22646b041e5e0941fbe361eccfe77a2a1e42f
---
 src/com/android/camera/PhotoModule.java             | 5 +++++
 src/com/android/camera/VideoModule.java             | 6 ++++++
 src/com/android/camera/WideAnglePanoramaModule.java | 5 +++++
 3 files changed, 16 insertions(+)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 9855332d2..78cfc81c6 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -2956,6 +2956,7 @@ public class PhotoModule
         }
         switch (keyCode) {
             case KeyEvent.KEYCODE_VOLUME_UP:
+            case KeyEvent.KEYCODE_MEDIA_NEXT:
                 if (mFirstTimeInitialized && (mUI.mMenuInitialized)) {
                     if (!CameraActivity.mPowerShutter && !CameraUtil.hasCameraKey()) {
                         onShutterButtonFocus(true);
@@ -2965,6 +2966,7 @@ public class PhotoModule
                 }
                 return true;
             case KeyEvent.KEYCODE_VOLUME_DOWN:
+            case KeyEvent.KEYCODE_MEDIA_PREVIOUS:
                 if (mFirstTimeInitialized && (mUI.mMenuInitialized)) {
                     if (!CameraActivity.mPowerShutter && !CameraUtil.hasCameraKey()) {
                         onShutterButtonFocus(true);
@@ -2982,6 +2984,7 @@ public class PhotoModule
                 }
                 return false;
             case KeyEvent.KEYCODE_CAMERA:
+            case KeyEvent.KEYCODE_HEADSETHOOK:
                 if (mFirstTimeInitialized && event.getRepeatCount() == 0) {
                     onShutterButtonClick();
                 }
@@ -3012,6 +3015,8 @@ public class PhotoModule
         switch (keyCode) {
             case KeyEvent.KEYCODE_VOLUME_UP:
             case KeyEvent.KEYCODE_VOLUME_DOWN:
+            case KeyEvent.KEYCODE_MEDIA_NEXT:
+            case KeyEvent.KEYCODE_MEDIA_PREVIOUS:
                 if (!CameraActivity.mPowerShutter && !CameraUtil.hasCameraKey()
                         && mFirstTimeInitialized) {
                     onShutterButtonClick();
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 9a6f20eaa..db0283ff2 100755
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -1390,6 +1390,7 @@ public class VideoModule implements CameraModule,
 
         switch (keyCode) {
             case KeyEvent.KEYCODE_VOLUME_UP:
+            case KeyEvent.KEYCODE_MEDIA_NEXT:
                 if (event.getRepeatCount() == 0 && !CameraActivity.mPowerShutter &&
                         !CameraUtil.hasCameraKey()) {
                     mUI.clickShutter();
@@ -1398,6 +1399,7 @@ public class VideoModule implements CameraModule,
                 }
                 return true;
             case KeyEvent.KEYCODE_VOLUME_DOWN:
+            case KeyEvent.KEYCODE_MEDIA_PREVIOUS:
                 if (event.getRepeatCount() == 0 && !CameraActivity.mPowerShutter &&
                         !CameraUtil.hasCameraKey()) {
                     mUI.clickShutter();
@@ -1406,6 +1408,7 @@ public class VideoModule implements CameraModule,
                 }
                 return true;
             case KeyEvent.KEYCODE_CAMERA:
+            case KeyEvent.KEYCODE_HEADSETHOOK:
                 if (event.getRepeatCount() == 0) {
                     mUI.clickShutter();
                 }
@@ -1434,16 +1437,19 @@ public class VideoModule implements CameraModule,
     public boolean onKeyUp(int keyCode, KeyEvent event) {
         switch (keyCode) {
             case KeyEvent.KEYCODE_VOLUME_UP:
+            case KeyEvent.KEYCODE_MEDIA_NEXT:
                 if (!CameraActivity.mPowerShutter && !CameraUtil.hasCameraKey()) {
                     mUI.pressShutter(false);
                 }
                 return true;
             case KeyEvent.KEYCODE_VOLUME_DOWN:
+            case KeyEvent.KEYCODE_MEDIA_PREVIOUS:
                 if (!CameraActivity.mPowerShutter && !CameraUtil.hasCameraKey()) {
                     mUI.pressShutter(false);
                 }
                 return true;
             case KeyEvent.KEYCODE_CAMERA:
+            case KeyEvent.KEYCODE_HEADSETHOOK:
                 mUI.pressShutter(false);
                 return true;
             case KeyEvent.KEYCODE_POWER:
diff --git a/src/com/android/camera/WideAnglePanoramaModule.java b/src/com/android/camera/WideAnglePanoramaModule.java
index d3c4ff163..4e0da9fbe 100644
--- a/src/com/android/camera/WideAnglePanoramaModule.java
+++ b/src/com/android/camera/WideAnglePanoramaModule.java
@@ -1255,8 +1255,11 @@ public class WideAnglePanoramaModule
         switch (keyCode) {
             case KeyEvent.KEYCODE_VOLUME_UP:
             case KeyEvent.KEYCODE_VOLUME_DOWN:
+            case KeyEvent.KEYCODE_MEDIA_NEXT:
+            case KeyEvent.KEYCODE_MEDIA_PREVIOUS:
                 return true;
             case KeyEvent.KEYCODE_CAMERA:
+            case KeyEvent.KEYCODE_HEADSETHOOK:
                 if (event.getRepeatCount() == 0) {
                     onShutterButtonClick();
                 }
@@ -1272,6 +1275,8 @@ public class WideAnglePanoramaModule
         switch (keyCode) {
             case KeyEvent.KEYCODE_VOLUME_UP:
             case KeyEvent.KEYCODE_VOLUME_DOWN:
+            case KeyEvent.KEYCODE_MEDIA_NEXT:
+            case KeyEvent.KEYCODE_MEDIA_PREVIOUS:
                 if (!CameraActivity.mPowerShutter && !CameraUtil.hasCameraKey()) {
                     onShutterButtonClick();
                 }
-- 
2.17.1

